### DISCUSSION TITLE
Adding "screen_id" via timelineVariable

### DISCUSSION THREAD

**Discussion by rMinion** [jspsych/jsPsych#1996] [Q&A] [answered]
Hi Guys, 

I'm trying to label my trials via timelineVariable. Is it possible or it will be messed up with the logic of "on_finish"? 

My code looks like this: 

```
var trials = 
{
  timeline:
  [
    {
      type: 'html-keyboard-response',
      data: {screen_id: jsPsych.timelineVariable(`data`)},
      stimulus: jsPsych.timelineVariable(`litera`),
      choices: jsPsych.timelineVariable(`choices`),
      trial_duration: jsPsych.timelineVariable(`trial_duration`),
      on_finish: jsPsych.timelineVariable(`on_finish`),       
         
    }
  ],
  timeline_variables: 
  [
    { 
      data: function() 
      {
        switch(wybor) 
        {
          case "a":
            var screen_id = `trial_a`;
            return screen_id;
          case "b":
            var screen_id = `trial_b`;
            return screen_id;
          case "c":
            var screen_id = `trial_c`;
            return screen_id;
        }
      },   
 ... OTHER      

```
I get the variable "sreen_id" in the csv output but it's empty.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @rMinion,

I think we need more context in the code to be helpful here. What is `wybor` and where is it defined?

**Reply by rMinion** [reply]
Thank you for answering. "wybor" It's a variable defined at the beginning of the script. It's tracking participants response ("a","b","c") during trials. It works well for all other timelineVariable calls (ex. stimulus, choices, trial_duration). I was thinking that maybe there is some order in which timelineVariables are called or that the execution of "on_finish" somehow overrides timelineVariables values.

**Reply by rMinion** [reply]
@jodeleeuw Sorry for littering the board! The code works as intended, all data records correctly! I was using excel for getting a glimpse for the data and it seems that the results instead of getting to corresponding columns were shifted in some strange way. I switched for JSON and convert it via R and it works perfectly.
